sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -164,9 +168,16 @@ def tune(
         self,
         # TODO (andreyvelich): How to be consistent with other APIs (name) ?
         name: str,
-        objective: Callable,
-        parameters: Dict[str, Any],
```

Comment:
I think, we should keep `parameters` for now. So user will have 2 options:
- Set `objective + parameters + base_image` or
- Set `model_provider_parameters + dataset_provider_parameters + trainer_parameters`.

I think, migration of `parameter` to something else, should be out of scope of this change.

WDYT @helenxie-bit @johnugeorge @tenzen-y @deepanker13 ?

Reply:
I currently use `trainer_parameters` for both options to avoid user confusion about which one to use for each option. I think it also makes sense if we keep `parameters` and make explanations about that.

Reply:
As we discussed on the call, we will keep `parameters` to avoid breaking changes for `tune` API in this PR.

Reply:
That makes sense. I keep `parameters` in `tune` API, and when users set a custom objective function, they will use `parameters` to define the search space for hyperparameters.